File: src/main/java/io/myosotisdev/utopianism/modules/dungeon/Dungeon.java
package io.myosotisdev.utopianism.modules.dungeon;

import io.myosotisdev.utopianism.modules.region.Region;
import io.myosotisdev.utopianism.modules.player.ImplPlayer;
import net.minestom.server.instance.Instance;

import java.util.List;

/**
 *
 */
public class Dungeon
{
    private DungeonRecord record;
    private Region        newRegion;
    private Instance      world;

    private List<ImplPlayer> players;

    private volatile long countdown;
}

File: src/main/java/io/myosotisdev/utopianism/modules/dungeon/Dungeon.kt
package io.myosotisdev.utopianism.modules.dungeon

import io.myosotisdev.utopianism.modules.player.PlayerEx
import io.myosotisdev.utopianism.modules.region.Region
import net.minestom.server.instance.Instance

/**
 *
 */
class Dungeon(val record: DungeonRecord, val region: Region, val world: Instance)
{
    val players: List<PlayerEx> = ArrayList()

    @Volatile
    var countdown: Long = 0
}